### PR TITLE
Jupyterlab 4 upgrade

### PR DESCRIPTION
**Pull request by peytondmurray**
This PR adds Jupyterlab 4 support for `jupyterlab-skip-traceback`. I tested this out with 4.0.0b1 and everything seemed to work:

image

There are some minor cosmetic changes I noticed from 3.x due to changes in global styles; the most noticeable one that I needed to fix was the size of the copy button, which appeared much larger in 4.x than in 3.x. Aside from that there's a slight difference in transparency of the caret and the copy buttons - I'm hoping that going with the new 4.x styles is acceptable, but if not I can make the change to match the old styles.

### CODE REVIEW

**Review by singharpit94 — COMMENTED** [review]
I have added few comments. Couple of important things which is blocking me to test out the changes locally

I am not able to run` pip install -e .` without a setup.py file, it complains with the below error:
```
ERROR: File "setup.py" not found. Directory cannot be installed in editable mode: /codemill/singhar/jupyterlab-skip-traceback
(A "pyproject.toml" file was found, but editable mode currently requires a setup.py based build.)
```

Also when I run `pip install .` (without editable mode install), I run into the below error
`ValueError: Extensions require a devDependency on @jupyterlab/builder@^3.6.2, you have a dependency on 4.0.0-rc.0`

I checked my node_modules in my venv and it does has @jupyterlab/builder@ 4.0.0-rc.0, do I need to upgrade some other packages to get this working, any leads here?

File: .github/workflows/build.yml
Comment (peytondmurray) [reply .github/workflows/build.yml:46]: The artefacts just used in the `test_isolated` step. This could in principle be condensed into a single job, but it's the default for the `ts-extension-cookiecutter` repo, so I kept it. It's really up to you if you want to get rid of `test_isolated` :shrug:

File: package.json
Comment (peytondmurray) [reply package.json:114]: It's definitely not mandatory, and there's no strong reason to do so other than reducing the amount of _stuff_ cluttering up the repository. I'm not a big fan of needing loads and loads of boilerplate files clogging the repo as is the usual for the extension cookiecutter, but if you prefer these to be broken out into separate files then I'm happy to do so.

File: src/index.ts
Comment (peytondmurray) [reply src/index.ts]: Sure, I'll set the appropriate rule for prettier.

**Comment by peytondmurray** [comment]
As far as your build issues go:

1. What version of everything are you using? I think editable builds have been in this build backend for a while. Can you try using a more recent toolchain?
2. 99% sure this is something to do with you having old build stuff interfering with the current build. Do a `git clean -fdx` to clear everything out. Then start from a new virtualenv and try building again.

**Comment by singharpit94** [comment]
> As far as your build issues go:
> 
> 1. What version of everything are you using? I think editable builds have been in this build backend for a while. Can you try using a more recent toolchain?
> 2. 99% sure this is something to do with you having old build stuff interfering with the current build. Do a `git clean -fdx` to clear everything out. Then start from a new virtualenv and try building again.

Yes, I was able to use the upgraded extension after fixing build issues at my end.

**Review by singharpit94 — APPROVED** [review]

**Comment by peytondmurray** [comment]
@singharpit94 @ankit-gautam23 I think this is ready to be merged. Is there anything else left to do here?

**Comment by ankit-gautam23** [comment]
@peytondmurray build issue was the biggest concern as @singharpit94 has solved it so I think we are good to go.

**Comment by peytondmurray** [comment]
@ankit-gautam23 or @singharpit94 I don't have merge rights, can you approve the workflow and merge when it passes?

**Comment by singharpit94** [comment]
Good from my end as well. One minor thing, can you update the README too with the latest changes, post that we can merge

**Comment by singharpit94** [comment]
Good from my end as well. One minor thing, can you update the README too with the latest changes, post that we can merge

**Comment by singharpit94** [comment]
@peytondmurray I think we will need some changes wrt to bumping the version in package.json, changelog. Also few tests are failing, can you take a look?

**Comment by peytondmurray** [comment]
Oh interesting - the link checker already caught some failing links to invalid tags. Nice!

**Comment by peytondmurray** [comment]
@singharpit94 Okay, `README.md`, `package.json`, and `CHANGELOG.md` were modified to bump the version and fix the link test.

**Comment by singharpit94** [comment]
@peytondmurray check links still failing, do we need to release first or update somewhere else to get this build green?

**Comment by peytondmurray** [comment]
> @peytondmurray check links still failing, do we need to release first or update somewhere else to get this build green?

For the `CHANGELOG.md` failure in principle yes, because the 5.0.0 release has not been tagged yet. Or we could just update `CHANGELOG.md` after we merge this PR.

For the `README.md` failure, it looks like something is wrong with the binder link. Not sure what to do about this, I'm getting just a generic 500 internal server error so I assume this is an issue with mybinder.org right now.

**Comment by peytondmurray** [comment]
Huh, so the extension template used in the upgrade incorrectly generates the wheel names for the github build action when uploading/downloading artifacts. I'll report this as a bug on the jupyterlab repo.

**Comment by peytondmurray** [comment]
@singharpit94 @ankit-gautam23 Looks like tests are now passing; after we merge this I'll make another PR to update `CHANGELOG.md`.

**Pull request merged**
